test/Common.cpp

```diff
@@ -222,4 +223,19 @@ void CommonOptions::setSingleton(std::unique_ptr<CommonOptions const>&& _instanc
 
 std::unique_ptr<CommonOptions const> CommonOptions::m_singleton = nullptr;
 
+bool isValidSemanticTestPath(boost::filesystem::path const& _path)
+{
+	bool insideSemanticTests = false;
+	fs::path parent_path;
+	for (auto const& element: _path)
+	{
+		parent_path = parent_path / element;
+		if (boost::ends_with(canonical(parent_path).generic_string(), "test/libsolidity/semanticTests"))
```

Comment:
I think you should ensure that there's either `/` before `test/` or it's the beginning of the string.